src/test/java/teammates/test/BaseComponentTestCase.java

```diff
@@ -115,23 +114,25 @@ protected void removeAndRestoreTypicalDataBundle() {
     }
 
     @Override
-    protected String doRemoveAndRestoreDataBundle(DataBundle dataBundle) {
+    protected boolean doRemoveAndRestoreDataBundle(DataBundle dataBundle) {
         try {
             logic.removeDataBundle(dataBundle);
             logic.persistDataBundle(dataBundle);
-            return Const.StatusCodes.BACKDOOR_STATUS_SUCCESS;
+            return true;
         } catch (Exception e) {
-            return Const.StatusCodes.BACKDOOR_STATUS_FAILURE + ": " + TeammatesException.toStringWithStackTrace(e);
+            System.out.println(TeammatesException.toStringWithStackTrace(e));
+            return false;
         }
     }
 
     @Override
-    protected String doPutDocuments(DataBundle dataBundle) {
+    protected boolean doPutDocuments(DataBundle dataBundle) {
         try {
             logic.putDocuments(dataBundle);
-            return Const.StatusCodes.BACKDOOR_STATUS_SUCCESS;
+            return true;
         } catch (Exception e) {
```

Comment:
This is not related to this PR but catching all `Exception` is [rarely a good choice](https://stackoverflow.com/a/2416334/13750147). It is much better to catch the specific ones that you know will be thrown and handle them.

Reply:
Rarely does not mean always. In this case:
- This is a test only method. That on its own doesn't justify to be lazy and catch all `Exception`s, but the next reason will/should.
- Any kind of failure during persisting search documents (or test data, because the methodology of catching-all-`Exception`s are the same) warrant the same handling: retry or halt (depending on number of attempts), i.e. no special case needs to be handled differently. If each case is caught separately but in the end everything is handled the same, that becomes a case of "doing it just for the sake of doing it", which is an anti-pattern.